outputs/quickwit.go

```diff
@@ -0,0 +1,158 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package outputs
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/falcosecurity/falcosidekick/types"
+)
+
+type QuickwitDynamicMapping struct {
+	Description string `json:"description"`
+	Fast        bool   `json:"fast"`
+	ExpendDots  bool   `json:"expand_dots"`
+	Indexed     bool   `json:"indexed"`
+	Record      string `json:"record"`
+	Stored      bool   `json:"bool"`
+	Tokenizer   string `json:"tokenizer"`
+}
+
+type QuickwitFieldMapping struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
+	Fast bool   `json:"fast"`
+}
+
+type QuickwitSearchSettings struct {
+	DefaultSearchFields []string `json:"default_search_fields"`
+}
+
+type QuickwitDocMapping struct {
+	DynamicMapping QuickwitDynamicMapping `json:"dynamic_mapping"`
+	FieldMappings  []QuickwitFieldMapping `json:"field_mappings"`
+	Mode           string                 `json:"mode"`
+	StoreSource    bool                   `json:"store_source"`
+	TimestampField string                 `json:"timestamp_field"`
+}
+
+type QuickwitMappingPayload struct {
+	Id             string                 `json:"index_id"`
+	Version        string                 `json:"version"`
+	SearchSettings QuickwitSearchSettings `json:"search_settings"`
+	DocMapping     QuickwitDocMapping     `json:"doc_mapping"`
+}
+
+func (c *Client) AutoCreateQuickwitIndex(args types.InitClientArgs) error {
+	config := args.Config.Quickwit
+	if !config.AutoCreateIndex {
+		return nil
+	}
+
+	endpointUrl := fmt.Sprintf("%s/%s/indexes", config.HostPort, config.ApiEndpoint)
+	quickwitInitClient, err := InitClient("QuickwitInit", endpointUrl, config.MutualTLS, config.CheckCert, args)
+	if nil != err {
+		return err
+	}
+
+	mapping := &QuickwitMappingPayload{
+		Id:      config.Index,
+		Version: config.IndexVersion,
+		DocMapping: QuickwitDocMapping{
+			Mode:           "dynamic",
+			StoreSource:    true,
+			TimestampField: "time",
+			DynamicMapping: QuickwitDynamicMapping{
+				Description: "Falco",
+				Fast:        true,
+				ExpendDots:  true,
+				Indexed:     true,
+				Stored:      true,
+				Record:      "basic",
+				Tokenizer:   "raw",
+			},
+			FieldMappings: []QuickwitFieldMapping{
+				{
+					Name: "time",
+					Type: "datetime",
+					Fast: true,
+				},
+				{
+					Name: "priority",
+					Type: "i64",
+					Fast: true,
+				},
+				{
+					Name: "source",
+					Type: "text",
+					Fast: true,
+				},
+				{
+					Name: "output",
+					Type: "string",
+					Fast: true,
+				},
+				{
+					Name: "rule",
+					Type: "string",
+					Fast: true,
+				},
+			},
+		},
+		SearchSettings: QuickwitSearchSettings{
+			DefaultSearchFields: []string{"source", "output"},
+		},
+	}
+
+	err = quickwitInitClient.Post(mapping)
+
+	// This error means it's an http 400 (meaning the index already exists, so no need to throw an error)
+	if err.Error() == "header missing" {
```

Comment:
This is the only way to know if it's a 400 error or not.

But we're going to have error logs on each start if the user is keeping the config `auto_create_index` set to `true`. I wonder if we should re-implement the `sendRequest` method to make it more modular and overridable (for example being able to inject a logger that we can mock, or return the real payload to be able to read it)

@Issif @fmassot what's your preference to handle that in a better way ?

Reply:
If you change `sendRequest` to be able to return the real payload, it implies you to change the code of a lot of outputs, I admit it's a pain. This method became ugly with time, by adding layers for some specific outputs.
Do you have any other method to check first if the index is present?

Reply:
I can do a GET before (done with this commit: 5bd04771abcc8fc7e98556ee4aa7c29e04240c04) but same result, I have a log error for no reason.

```shell
falco-test-falco-sidekick-1  | 2024/01/17 11:13:48 [ERROR] : QuickwitCheckAlreadyExists - resource not found (404): {
falco-test-falco-sidekick-1  |   "message": "index `falco` do(es) not exist"
falco-test-falco-sidekick-1  | }
falco-test-falco-sidekick-1  | 2024/01/17 11:13:48 [DEBUG] : Quickwit mapping - &{falco 0.7 {[source output]} {{Falco true true true basic true raw} [{time datetime true} {priority i64 true} {source text true} {output text true} {rule text true}] dynamic true time}}
falco-test-falco-sidekick-1  | 2024/01/17 11:13:49 [INFO]  : QuickwitInit - Post OK (200)
```

Reply:
The version with a `GET /describe` before is a little bit better because we'll have an error log only on the first init of the index and not on every start of a falcosidekick instance. If I restart my container:

```shell
falco-test-falco-sidekick-1  | 2024/01/17 11:22:00 [INFO]  : QuickwitCheckAlreadyExists - Post OK (200)
falco-test-falco-sidekick-1  | 2024/01/17 11:22:00 [INFO]  : Falcosidekick version: a817fb2
falco-test-falco-sidekick-1  | 2024/01/17 11:22:00 [INFO]  : Enabled Outputs : [Quickwit]
falco-test-falco-sidekick-1  | 2024/01/17 11:22:00 [INFO]  : Falcosidekick is up and listening on :2801
```

I flag this PR as ready to be reviewed @Issif I think it's "good enough" and if you want to refactor the `sendRequest` we can do it on a dedicated pull request?

Reply:
I left some comments and suggestions. Seems good to me with this implementation too. For `sendRequest` it requires another PR for sure.

Reply:
@Issif All your comments and suggestions have been implemented and re-tested again : still working fine :)

Thanks for those!